### PR TITLE
35-add-save-model-function

### LINKED ISSUE #35: Add a way to save the models
### The problem
- Add a way to save the models
### The solution / the feature
- Add a save function which can be called at a intervall or given reward or given epoch
### Which Pros/Cons does this solution have?
- We can do the unsupervised pre-training once and just reuse this saved model for testing

### PR DESCRIPTION
**Pull request by Wizzardexe**
**Issue:** Fixes #35
Add a way to save the models
**What?**
- Added save/load function for model and replay buffer

**Ready to merge to main?**
- [ ] Linter passes
- [ ] PR Approved

### CODE REVIEW

**Review by flyingpalli — CHANGES_REQUESTED** [review]
Looks good to me. It might be useful to also save models/buffers when the program is stopped with a KeyboardInterrupt. I know that this might generate a lot of data when you debug, but I still feel like this could prevent some frustration... What do you think?

**Comment by Wizzardexe** [comment]
Yeah, that sounds like a good idea. I will try to implement that.

File: framework/sac_rlhf.py [hunk 1]
```diff
@@ -176,6 +183,57 @@ def update_target_networks(source_net, target_net, tau):
     for param, target_param in zip(source_net.parameters(), target_net.parameters()):
         target_param.data.copy_(tau * param.data + (1 - tau) * target_param.data)
 
+def save_model_all(run_name: str, step: int,  state_dict: dict):
+    model_folder = f"./models/{run_name}/{step}"
+    if not os.path.exists(model_folder):
```
Comment (bkuen) [inline framework/sac_rlhf.py:188]: Just a cool shortcut: `os.makedirs(path, exist_ok=True)`

**Review by flyingpalli — APPROVED** [review]
